### PR TITLE
sysdig: 0.10.0 -> 0.11.0

### PR DESCRIPTION
**Pull request by markus1189**
Update sysdig to 0.11.0.  Also correctly set the version during build.

### CODE REVIEW

**Comment by grahamc** [comment]
That didn't take long to patch, unfortunately it doesn't seem to build with Chroot enabled due to networking being disabled.

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

*[pkgs/os-specific/linux/sysdig/default.nix, line 27 \[r1\]](https://reviewable.io:443/reviews/nixos/nixpkgs/17124#-KN7iL1eSNjj9acLFmNG:-KN7iL1eSNjj9acLFmNH:1577229231) ([raw file](https://github.com/nixos/nixpkgs/blob/794b93992d9eee08f6859a6f981478ae06832324/pkgs/os-specific/linux/sysdig/default.nix#L27)):*
> ```Nix
>   cmakeFlags = [
>     "-DUSE_BUNDLED_DEPS=OFF"
>     "-DUSE_BUNDLED_JQ=ON"
> ```

With this enabled, it seems to try and download fromthe internet during the build, which isn't allowed:


```
Scanning dependencies of target jq
[  1%] Creating directories for 'jq'
[  2%] Performing download step (download, verify and extract) for 'jq'
-- downloading...
     src='http://download.draios.com/dependencies/jq-1.5.tar.gz'
     dst='/tmp/nix-build-sysdig-0.11.0.drv-0/sysdig-0.11.0/build/jq-prefix/src/jq-1.5.tar.gz'
     timeout='none'
CMake Error at jq-stamp/download-jq.cmake:27 (message):
  error: downloading 'http://download.draios.com/dependencies/jq-1.5.tar.gz'
  failed

    status_code: 6
    status_string: "Couldn't resolve host name"
    log: getaddrinfo(3) failed for download.draios.com:80

  Couldn't resolve host 'download.draios.com'

  Closing connection 0
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/nixos/nixpkgs/17124#-:-KN7hyzOzHRczuZZnfPI:546775244)*

**Comment by markus1189** [comment]
Oh I didn't take that into account sorry ;)

What would be the best way to fix this?  The problem is that we can't use nixpkgs' `jq` because sysdig relies on it's own version with some headers which aren't installed with standard `jq`.  Adding a separate package for `sysdig jq` is not a solution.  Can I somehow specify to download it in the nix expression?

**Comment by grahamc** [comment]
You possibly could. Do another fetchurl and in preBuild maybe or preConfigure `cp` it to where sysdig is expecting it to live.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/nixos/nixpkgs/17124#-:-KN86p4PoB8AUx8x2TJx:-1051218068)*

**Comment by markus1189** [comment]
Okay I added another `fetchurl` but it somehow still fails.  How can I find out why it fails?  E.g. https://travis-ci.org/NixOS/nixpkgs/jobs/146186286 doesn't seem to contain an error...

I tried locally with `nix-build -v --option build-use-chroot true -A sysdig` and that works.

**Comment by grahamc** [comment]
It certainly builds and works for me. I'd like another person to review it, because it isn't consistent with other builds.

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/nixos/nixpkgs/17124#-:-KN8f34wN1S3EWh6WoxX:-471065434)*

**Comment by grahamc** [comment]
I had @joachifm check it out. Would you put a note in the code explaining why we use this other version, link to the issue in sysdig, etc? For future maintainers.

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/nixos/nixpkgs/17124#-:-KN8g3xW8GFDuIJLSdEJ:-2004013531)*

File: pkgs/os-specific/linux/sysdig/default.nix [hunk 1]
```diff
@@ -3,14 +3,28 @@ let
...
+  # sysdig-0.11.0 depends on some headers from jq which are not
+  # installed by it.  Sysdig allows the use of a bundled jq version,
+  # from which it would normally get the headers, but this requires
+  # network during the build which is not available with chroot in
+  # nix.
+  #
+  # As a workaround, we enable the bundled jq from sysdig, but
+  # download the jq archive prior to the build, such that it can work
+  # under chroot.
+  # Relevant sysdig issue: https://github.com/draios/sysdig/issues/626
```
Comment (markus1189) [inline pkgs/os-specific/linux/sysdig/default.nix]: Is that okay as a description?

File: pkgs/os-specific/linux/sysdig/default.nix [hunk 2]
```diff
@@ -3,14 +3,28 @@ let
   inherit (stdenv.lib) optional optionalString;
   s = rec {
     baseName="sysdig";
-    version = "0.10.0";
+    version = "0.11.0";
     name="${baseName}-${version}";
     url="https://github.com/draios/sysdig/archive/${version}.tar.gz";
```
Comment (joachifm) [inline pkgs/os-specific/linux/sysdig/default.nix]: Looks like this ends up being passed to `fetchurl`; please consider `fetchFromGitHub` instead, it'll be more stable wrt. to changes in the way github generates the archive.

File: pkgs/os-specific/linux/sysdig/default.nix (hunk 1 above)
Comment (joachifm) [inline pkgs/os-specific/linux/sysdig/default.nix]: I think so. Just mentioning the header issue would be sufficient, though (that builds cannot have network access is implied by purity, even if enforcement is optional).

**Comment by markus1189** [comment]
Changed it to use `fetchFromGitHub` for sysdig.  I had to leave the `fetchurl` for `jq-prefix` because I actually need the `jq-1.5.tar.gz` file in order to not trigger the bundled jq download from sysdig.

**Comment by markus1189** [comment]
Also, rebased and shortened the description

**Comment by grahamc** [comment]
Thank you!

**Pull request merged**